sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1837,6 +1844,39 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Get previous application versions for an application.
+   * @param applicationId application id.
+   * @param options method options.
+   * - `limit` number of previous application versions to be returned.
+   * - `loadDynamicData` optionally loads the dynamic data.
+   * @returns previous application versions.
+   */
+  async getPreviousApplicationVersions(
+    applicationId: number,
+    options?: { loadDynamicData?: boolean; limit?: number },
+  ): Promise<Application[]> {
+    const application = await this.repo.findOne({
+      select: { applicationNumber: true, submittedDate: true },
+      where: { id: applicationId },
+    });
+    return this.repo.find({
+      select: {
+        id: true,
+        submittedDate: true,
+        data: !!options?.loadDynamicData as unknown,
+      },
+      where: {
+        submittedDate: LessThan(application.submittedDate),
+        applicationNumber: application.applicationNumber,
+      },
+      order: {
```

Comment:
i think for now, the ordering of the application according to the submitted date will help us fetch the previous version, don't we need to think of having versioning of the application at this point? Thoughts??

Reply:
If by version do you mean something like what was applied to offer, yes, it would be good and it was always the idea but we never went for it. Either way, it's not an immediate issue for the current sprint tickets.